lib/defaults/defaults.go

```diff
@@ -323,6 +323,10 @@ const (
 	// DefaultRedisUsername is a default username used by Redis when
 	// no name is provided at connection time.
 	DefaultRedisUsername = "default"
+
+	// TLSCertificatesCacheReloadInterval is the period after which the in-memory cache of TLS certificates
+	// is refreshed from disk.
+	TLSCertificatesCacheReloadInterval = 24 * time.Hour
```

Comment:
Instead of use hardcoded interval it would be better to subscribe to [inotify](https://man7.org/linux/man-pages/man7/inotify.7.html) events.

Golang example: https://pkg.go.dev/k8s.io/utils/inotify

Reply:
This package seems to be depreciated, the documentation recommends using https://github.com/fsnotify/fsnotify instead

I've attempted an experimental implementation of this, but it appears to be much more complex than expected with **[many edge-cases](https://github.com/fsnotify/fsnotify/issues/372)** that must be handled

I'm willing to move forward if there's some consensus (ideally from the maintainers) around this approach, but I'm wary of wasting my time on this if the result is unlikely to be merged

CC: @r0mant @russjones @zmb3, sorry for the pings, what are your thoughts on this?

Reply:
My previous comment stands. I think there are a number of possible solutions to address this problem, and I'm not convinced the one implemented in this PR is the best option (it may well be, we just haven't had any discussion)

If you're still interested in working on this, I would recommend continuing the discussion in #3815 with a more detailed proposal.